### PR TITLE
restructure library exports

### PR DESCRIPTION
**Pull request by crfeliz**

### CODE REVIEW

File: src/indexExample.ts [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
-  FYI, if it's running correctly, your first logged output in the console will likely say "ERROR". Sorry about that ;-)
-       You'll know it's run correctly if you have a new file exampleData/analysisOutputRaw.json with some GitHub issues in it.
+    FYI, if it's running correctly, your first logged output in the console
+    will likely say "ERROR". Sorry about that ;-)
+        You'll know it's run correctly if you have a new file
+        exampleData/output.json with some GitHub issues in it.
 */
+
+import * as mariner from './mariner/index'; // This is used during development
+// import * as mariner from 'oss-mariner'    // This is how it the npm package would normally be used
```
Comment (crfeliz) [inline src/indexExample.ts:18]: remove the word it

**Comment by kevindigo** [comment]
I updated the indexExample that Carlos started. It now has reasonable defaults for input and output paths, so you only have to add the github token to the environment. I also renamed the github token environment variable to make it more plausible to set it in .profile or similar. 

I also updated the comments to hopefully be more helpful to anyone using it as an example.

File: src/indexExample.ts [hunk 2]
```diff
@@ -35,8 +47,12 @@ class FancyLogger implements Logger {
...
+const logger = new FancyLogger();
+mariner.setLogger(logger);
+
+logger.info(`Input:  ${inputFilePath}`);
+logger.info(`Output: ${outputFilePath}`);
 
-const ddr = new DependencyDetailsRetriever();
+// Here is the code to actually call mariner
+const ddr = new mariner.DependencyDetailsRetriever();
 ddr.run(token, inputFilePath, outputFilePath);
```
Comment (crfeliz) [inline src/indexExample.ts:58]: Perhaps we should add `.then(() => logger.info('Done!')` here to make it clear run returns a promise.

**Comment by crfeliz** [comment]
I can't find a way to add a proper review. Nevertheless I approve the changes Kevin made.

**Pull request merged**